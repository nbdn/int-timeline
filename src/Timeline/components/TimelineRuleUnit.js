// @flow
import React from 'react';

type Props = {
  iUnit: number,
  unit: number | string,
  unitWidth: number
};

const TimelineRuleUnit = (props: Props) => {
  const { iUnit, unit, unitWidth } = props;
  return (
    <div
      key={`unit_${iUnit}`}
      className="TimelineRule__Item"
      style={{ width: unitWidth }}
    >
      <span>{unit}</span>
    </div>
  );
};

TimelineRuleUnit.defaultProps = {
  iUnit: 0,
  unit: '2000',
  unitWidth: 0
};

export default TimelineRuleUnit;
