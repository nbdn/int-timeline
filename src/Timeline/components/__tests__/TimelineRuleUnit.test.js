import React from 'react';
import { mount, shallow } from 'enzyme';
import TimelineRuleUnit from '../TimelineRuleUnit';

describe('TimelineRuleUnit component test', () => {
  let wrapper;
  let emptyPropsWrapper;
  beforeEach(() => {
    wrapper = shallow(
      <TimelineRuleUnit iUnit={0} unit={'2001'} unitWidth={100} />
    );
    emptyPropsWrapper = shallow(<TimelineRuleUnit />);
  });

  describe('component rendering right', () => {
    it('render props right match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('render with empty props match snapshot', () => {
      expect(emptyPropsWrapper).toMatchSnapshot();
    });
  });

  describe('component has right props', () => {
    it('has right default props', () => {
      const mountedWrapper = mount(<TimelineRuleUnit />);
      expect(mountedWrapper.prop('iUnit')).toEqual(0);
      expect(mountedWrapper.prop('unit')).toEqual('2000');
      expect(mountedWrapper.prop('unitWidth')).toEqual(0);
    });

    it('has right given props', () => {
      const mountedWrapper = mount(
        <TimelineRuleUnit iUnit={0} unit={'2001'} unitWidth={100} />
      );
      expect(mountedWrapper.prop('iUnit')).toEqual(0);
      expect(mountedWrapper.prop('unit')).toEqual('2001');
      expect(mountedWrapper.prop('unitWidth')).toEqual(100);
    });
  });

  describe('component structure', () => {
    it('has right wrapper class name', () => {
      expect(wrapper.hasClass('TimelineRule__Item')).toBeTruthy();
    });
  });
});
