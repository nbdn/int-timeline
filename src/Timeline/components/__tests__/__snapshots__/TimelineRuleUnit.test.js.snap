// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineRuleUnit component test component rendering right render props right match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TimelineRuleUnit
    iUnit={0}
    unit="2001"
    unitWidth={100}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "unit_0",
    "nodeType": "host",
    "props": Object {
      "children": <span>
        2001
</span>,
      "className": "TimelineRule__Item",
      "style": Object {
        "width": 100,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "2001",
      },
      "ref": null,
      "rendered": "2001",
      "type": "span",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "unit_0",
      "nodeType": "host",
      "props": Object {
        "children": <span>
          2001
</span>,
        "className": "TimelineRule__Item",
        "style": Object {
          "width": 100,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "2001",
        },
        "ref": null,
        "rendered": "2001",
        "type": "span",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`TimelineRuleUnit component test component rendering right render with empty props match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TimelineRuleUnit
    iUnit={0}
    unit="2000"
    unitWidth={0}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "unit_0",
    "nodeType": "host",
    "props": Object {
      "children": <span>
        2000
</span>,
      "className": "TimelineRule__Item",
      "style": Object {
        "width": 0,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "2000",
      },
      "ref": null,
      "rendered": "2000",
      "type": "span",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "unit_0",
      "nodeType": "host",
      "props": Object {
        "children": <span>
          2000
</span>,
        "className": "TimelineRule__Item",
        "style": Object {
          "width": 0,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "2000",
        },
        "ref": null,
        "rendered": "2000",
        "type": "span",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
